dotCMS/src/main/java/com/dotmarketing/beans/Inode.java

```diff
@@ -518,19 +518,7 @@ public boolean hasChildren() {
 	 */
 	public boolean hasParents() {
 
-		HibernateUtil dh = new HibernateUtil();
-		String query = "select count(*) from " + Inode.class.getName()
-				+ "  inode where inode.inode in (select tree.parent from "
-				+ Tree.class.getName() + "  tree where tree.child = "
-				+ this.inode + ")";
-		List results = new ArrayList();
-		try {
-			dh.setQuery(query);
-			results = dh.list();
-		} catch (DotHibernateException e) {
-			Logger.error(this.getClass(), e.getMessage(), e);
-		}
-		return ((Integer) results.get(0)).intValue() > 0;
+	    return TreeFactory.getTreesByChild(this.inode).size()>0;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)